pkg/agentconfig/container.go

```diff
@@ -26,15 +26,39 @@ func (a *ContainerBuilder) AgentContainer(ctx context.Context) (*core.Container,
 	ports := make([]core.ContainerPort, 0, 5)
 	confCns := a.configuredContainers(ctx)
 
+	names := make(map[string]int)
 	a.eachConfiguredContainer(confCns, func(app *core.Container, cc *Container) {
 		switch cc.Replace {
 		case ReplacePolicyContainer:
 			// Simply inherit the ports of the replaced container
 			ports = append(ports, app.Ports...)
 		case ReplacePolicyIntercept:
 			for _, ic := range PortUniqueIntercepts(cc) {
+				name := ic.ContainerPortName
+
+				// We don't want to apply duplication logic to empty strings
+				// as - is not a valid starting character for port names.
+				if name != "" {
+					if _, ok := names[name]; ok {
+						// if name already exists, append a number to it
+						names[ic.ContainerPortName]++
+						// convert number to string
+						number := "-" + strconv.Itoa(names[ic.ContainerPortName])
+						// if string length of name plus number is greater than 15
+						if len(ic.ContainerPortName)+len(number) > 15 {
+							// truncate name to 15 characters
+							name = ic.ContainerPortName[:15-len(number)] + number
+						} else {
+							name = ic.ContainerPortName + number
+						}
+					} else {
+						name = ic.ContainerPortName
+						names[ic.ContainerPortName] = 1
+					}
```

Comment:
I'm aware that this is a nit, but I believe this can be simplified further:
- The `name` is already assigned the value of `ic.ContainerPortName`, and can be used consistently in the rest of the logic. 
- Instead of discarding the number on the first lookup, it, it can be increased and just stored. This saves two hash lookups. Currently your `names[ic.ContainerPortName]++` construct will perform the same lookup again, and then a third lookup is made when converting the number to a string.
```suggestion
					if n, ok := names[name]; ok {
						// if name already exists, append a number to it
						n++
						names[name] = n
						// convert to numeric name suffix
						suffix := "-" + strconv.Itoa(n)
						// if string length of name plus suffix is greater than 15
						if len(name)+len(suffix) > 15 {
							// truncate name to 15 characters
							name = name[:15-len(suffix)]
						}
						name += suffix
					} else {
						names[name] = 1
					}
```